src/dvclive/live.py

```diff
@@ -506,13 +515,23 @@ def log_image(
             raise InvalidImageNameError(name)
 
         if name in self._images:
-            image = self._images[name]
+            image = self._images[name]["image"]
         else:
             image = Image(name, self.plots_dir)
-            self._images[name] = image
+            self._images[name] = {"image": image}
 
         image.step = self.step
         image.dump(val)
+
+        if annotations:
+            if not Annotations.could_log(annotations):
+                raise InvalidDataTypeError(name, type(annotations))
```

Comment:
Do you think we should actually raise an error? I know that's what we otherwise do if the image format is invalid, but we also tend to be cautious in raising errors in dvclive since it could potentially interrupt more important and long-running model training operations.

Reply:
Good point.

From my experience there is no good answer. I see two mood for datascientists:
 - the dev mood, where you're working on the training pipeline. You're launching small experiments with very fast iterations to fix problems on your data/training pipeline. In that case you want the training to crash and raise an error.
 - the production mood where you're working on a model and you don't want to touch the data/training pipeline. You're just changing parameters and running longs (hours/days) experiments. In this setup, I would be mad that the training crash at the end because my logger received a float instead of an int. This is quite easy to catch on the "dev mood" but there is some bugs that occurs very rarely like : the model didn't predict any object on the image, the bounding box have negative coordinates... In the production mode, I would like the logger to raise a warning. 
 
Based on this experience, and if we need to make a choice, I prefer DVClive to always behave in a production mood because this is where you have money at stake. 
There is only one downside about warning: datascientists don't read them. Their pytorch/tensorflow framework is already spamming with logs. Useful logs (for data scientists) like epoch number, the time it takes to complete an epoch, loss values...  But also a LOT of useless logs (for datascientists) about optimization using this env or this, deprecated function that one of your lib is using. Now if you had the logs cuda and co. are adding to optimize this kernel function... You get ~50 to 100 lines at the beginning of every training that nobody reads.
So warning is cool, but I doubt people will look at the logs to see our warning during training.

One cool improvement I could see is to use the `end` method to make a recap:
 - you could have done this better using this function
 - on this step, for this image, the coordinates were negative. Here is what we received, and here is what we expect...
 These types of improvements would be useful and looked at if they are a the end of the log file and well formatted.

Reply:
Yes, 100% agree with what you wrote about how data scientists work. Anyway, fine to keep as is and revisit if someone complains.